src/NovelRT/Ecs/Audio/AudioSystem.cpp

```diff
@@ -264,64 +207,43 @@ namespace NovelRT::Ecs::Audio
                 }
             }
         }
-
-        _service->CheckSources();
     }
 
-    uint32_t AudioSystem::CreateAudio(std::string fileName, bool isMusic)
+    uint32_t AudioSystem::RegisterSound(std::string fileName)
     {
         if (_counter == UINT32_MAX)
         {
-            // add logging here
             return 0U;
         }
 
-        // 0 is not valid here since we're working with only positive numbers... not that anyone should be loading
-        // thousands of sounds.
-        uint32_t value = 0U;
-        if (isMusic)
-        {
-            auto asset = _resourceManagerPluginProvider->GetResourceLoader()->LoadAudioFrameData(fileName);
-            auto handle = _service->LoadMusic(asset.processedAudioFrames, asset.channelCount, asset.sampleRate);
-            if (_service->IsLoaded(handle))
-            {
-                _musicCache.insert({_counter, handle});
-                value = _counter;
-                _counter++;
-            }
-        }
-        else
-        {
-            auto asset = _resourceManagerPluginProvider->GetResourceLoader()->LoadAudioFrameData(fileName);
-            auto handle = _service->LoadSound(asset.processedAudioFrames, asset.channelCount, asset.sampleRate);
-            if (_service->IsLoaded(handle))
-            {
-                _soundCache.insert({_counter, handle});
-                value = _counter;
-                _counter++;
-            }
-        }
-
-        return value;
+        auto asset = _resourceManagerPluginProvider->GetResourceLoader()->LoadAudioFrameData(fileName);
+        auto handle = _mixer->SubmitAudioBuffer(
+            NovelRT::Utilities::Misc::Span<float>(asset.processedAudioFrames.data(), asset.processedAudioFrames.size()),
+            asset.channelCount, asset.sampleRate);
+        _soundCache.emplace(handle, asset);
+        // if (_mixer->IsLoaded(handle))
+        // {
+        //     _soundCache.insert({_counter, handle});
+        //     value = _counter;
+        _counter++;
+        // }
+        return handle;
     }
 
-    void AudioSystem::ChangeAudioVolume(AudioEmitterComponent emitter, float desiredVolume)
-    {
-        if (emitter.isMusic)
-        {
-            _service->SetMusicVolume(desiredVolume);
-        }
-        else
-        {
-            _service->SetSoundVolume(_soundCache.at(emitter.handle), desiredVolume);
-        }
-    }
+    // void AudioSystem::ChangeAudioVolume(AudioEmitterComponent emitter, float desiredVolume)
+    // {
+    //     if (emitter.isMusic)
+    //     {
+    //         _mixer->SetMusicVolume(desiredVolume);
+    //     }
+    //     else
+    //     {
+    //         _mixer->SetSoundVolume(_soundCache.at(emitter.handle), desiredVolume);
+    //     }
+    // }
```

Comment:
Dead code?

Reply:
Somewhat - I had to uncomment this and fix the underlying mixer call to the proper signature. Should be good now.